src/signMessage.c

```diff
@@ -137,6 +139,12 @@ void handle_sign_message_parse_message(volatile unsigned int *tx) {
         //*tx = set_result_sign_message();
         // THROW(ApduReplySuccess);
         UNUSED(tx);
+    }
+
+    if ((G_command.non_confirm || G_called_from_swap) &&
+        !(G_command.non_confirm && G_called_from_swap)) {
+        // Blind sign requested NOT in swap context
```

Comment:
Could you explain what non_confirm mode supposed to do with the swap feature? I'm afraid you are mixing it with N_storage.settings.allow_blind_sign option because of the comment. Non confirm is normally used for other instructions to ignore user approval and just perform the operation. And in such case nothing is displayed on the Ledger.

Reply:
`G_command.non_confirm` is set when the transaction is received with a `P1_NON_CONFIRM`
In a normal case we throw an error when a SIGN request receives a `P1_NON_CONFIRM `because we don't want to allow bypassing the user screen confirmation
In SWAP mode it's the opposite, we actually expect `P1_NON_CONFIRM`, the user has already validated the `amount / fees / target currency / etc` during the validation of the SWAP flow

The comment is indeed confusing, I'll rephrase